tapis-project/tapis-ui
# task/TUI-249 - jobs wizard

## Overview:

Wizard based workflow for job submission

## Related Github Issues:

- [TUI-249](https://github.com/tapis-project/tapis-ui/issues/249)

## Summary of Changes:

tapis-ui/_common/Wizard component

- accepts a series of steps and a "finish" button component that is displayed on the last page.
- displays completion status of each step
- disables "next" button if current page is not complete (although this can be bypassed by clicking on the step navigation header)
- displays custom "finish" button component on last page of wizard instead of "next" button

tapis-ui/components/Jobs/JobLauncher

- Uses wizard based workflow
- Abstracts sections of steps into individual step components
- All components use a shared form context

## Testing Steps:

1. Try modifying form fields to verify that validation prevents job navigation and job submission, etc.

## UI Photos:

## Notes:

- Requires updates to @tapis/tapis-typescript due to production changes to Apps, Jobs and Systems
- Requires refactoring and unit tests of functions that validate job submission (to be addressed in #242 )
- Requires fefactoring of FileInputSteps components (to be addressed in #242)
